kotlinx-coroutines-core/common/src/flow/internal/ChannelFlow.kt

```diff
@@ -126,13 +143,30 @@ internal abstract class ChannelFlowOperator<S, T>(
 internal class ChannelFlowOperatorImpl<T>(
     flow: Flow<T>,
     context: CoroutineContext = EmptyCoroutineContext,
+    val dropWhileBusy: Boolean = false,
     capacity: Int = Channel.OPTIONAL_CHANNEL
 ) : ChannelFlowOperator<T, T>(flow, context, capacity) {
-    override fun create(context: CoroutineContext, capacity: Int): ChannelFlow<T> =
-        ChannelFlowOperatorImpl(flow, context, capacity)
 
-    override suspend fun flowCollect(collector: FlowCollector<T>) =
-        flow.collect(collector)
+    override fun create(context: CoroutineContext, capacity: Int): ChannelFlow<T> {
+        return ChannelFlowOperatorImpl(
+            flow,
+            context,
+            dropWhileBusy = capacity != Channel.CONFLATED && dropWhileBusy,
```

Comment:
This code properly belongs to `update` function. That is where all trickly "fusion" of parameters is performance. Here is should simply create implementation with the given params. To get there, `create` function shall have `dropWhileBusy` parameter.

Reply:
In order to do that, should I overload the create function or change the signature in `ChannelFlowOperator`?

Reply:
You should just add `dropWhileBusy` boolean to `create`. It is all internal code that is not directly available to end-users of API.